lib/listen/adapter/darwin.rb

```diff
@@ -6,17 +6,43 @@ module Adapter
     # Adapter implementation for Mac OS X `FSEvents`.
     #
     class Darwin < Base
-      OS_REGEXP = /darwin(1.+)?$/i
+      OS_REGEXP = /darwin(1\d+)/i
 
       # The default delay between checking for changes.
       DEFAULTS = { latency: 0.1 }
 
+      BUNDLER_DECLARE_GEM = <<-EOS.gsub(/^ {8}/, '')
+        rb-fsevent > 0.9.4 no longer supports OS X 10.6 through 10.8.
+
+        Please add the following to your Gemfile to avoid polling for changes:
+          require 'rbconfig'
+          if RbConfig::CONFIG['target_os'] =~ /darwin(1[0-3])/i
+            gem 'rb-fsevent', '<= 0.9.4'
+          end
+      EOS
+
+      def self.usable?
+        match = OS_REGEXP.match(RbConfig::CONFIG['target_os']) or return false
+        darwin_version = match[1].to_i
+        return true if darwin_version >= 13  # darwin13 is OS X 10.9 Mavericks
+
+        # OS X 10.6 through 10.8 only work with an older rb-fsevent version
+        begin
+          require 'rb-fsevent'
+          fsevent_version = Gem::Version.new(FSEvent::VERSION)
+          fsevent0_9_4 = Gem::Version.new('0.9.4') # supports 10.6-10.8
+          raise 'version too new' if fsevent_version > fsevent0_9_4
```

Comment:
Instead of raising and catching, you can just return true here and show the version error only to match the case, e.g.

```ruby
return true if fsevent_version <= fsevent0_9_4
Kernel.warn BUNDLER_DECLARE_GEM
false
rescue LoadError
false
end
```

(That's because a LoadError shouldn't talk about version problems, or that will confuse users).